rfcs/20210415-gpu-only/README.md

```diff
@@ -0,0 +1,185 @@
+# Proposal on introducing GPU only configurations
+
+
+## Motivation
+oneDNN supports multiple CPU and GPU runtimes. While it's possible to build
+oneDNN only for a CPU runtime the GPU runtime always requires a CPU runtime to be
+specified. For some customers the size of the oneDNN library matters a lot therefore
+if the customers only need a GPU runtime then having a redundant CPU runtime
+becomes a problem. One of the customers is OpenVINO whose request is the main motivation
+for introducing an ability to build oneDNN only for GPU runtime.
+
+## Proposal
+The proposal is to introduce the following GPU only configurations:
+* gpu_dpcpp (`DNNL_CPU_RUNTIME=NONE DNNL_GPU_RUNTIME=DPCPP`)
```

Comment:
As I know, DPCPP runtime still requires OCL engine to compile ocl kernels. So will this build for DPCPP disable unused OCL engine parts (if such parts exist)?

Reply:
You are right about using OCL engine to compile OpenCL kernels. Recently, DPC++ compiler introduced [online compilation](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/OnlineCompilation/OnlineCompilation.asciidoc) feature that can be used to compile OpenCL code without creating an OCL engine. We haven't switched to it yet because there is a regression in compilation time. The DPC++ team is working to resolve the issue. As soon as the regression is fixed we will switch to the online compiler and try to disable unused OpenCL code infrastructure.

Reply:
Sounds good to me. And thanks for mentioning online compiler feature. Might be useful for OpenVINO as well.